agent/unit/unit_test.go

```diff
@@ -0,0 +1,59 @@
+package unit_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/v2/agent/unit"
+)
+
+func TestDependencyCoordinator(t *testing.T) {
+	t.Parallel()
+
+	t.Run("ForwardAndReverseEdges", func(t *testing.T) {
+		t.Parallel()
+
+		graph := unit.NewDependencyCoordinator()
+		unit1 := &unit.Unit{Name: "unit1", Status: unit.StatusPending}
+		unit2 := &unit.Unit{Name: "unit2", Status: unit.StatusPending}
+		unit3 := &unit.Unit{Name: "unit3", Status: unit.StatusPending}
+		graph.AddEdge(unit1, unit2, unit.StatusCompleted)
+		graph.AddEdge(unit1, unit3, unit.StatusStarted)
```

Comment:
readability, non-blocking: if we were able to define these in DOT format, we could render them to e.g. PNG and then assert the hash of the result.

Reply:
This was a great idea. gonum has a toDOT mechanism. I've elected to maintain and check against fixture files, but the concept holds. Thanks!